lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -553,9 +553,16 @@ class DxilLowerCreateHandleForLib : public ModulePass {
     // Make sure no select on resource.
     bChanged |= RemovePhiOnResource();
 
-    if (m_bIsLib || m_bLegalizationFailed)
+    if (m_bLegalizationFailed)
       return bChanged;
 
+    if (m_bIsLib) {
+      if (DM.GetOP()->UseMinPrecision())
+        UpdateStructTypeForLegacyLayout();
+
+      return true;
```

Comment:
Why not return `bChanged` here?

Reply:
It's been almost two months and I don't remember. I made the change at the same time that I changed the prefix ( d5ada27a2a ). 

I don't think just returning `bChanged` is right. If the type names have changed, I think we need to return true. `UpdateStructTypeforLegacyLayout` doesn't currently indicate if it changed anything or not. I think I'll make it tell what it did and return `bChanged || bLocalChanged` where `bLocalChanged` is the result.

Does that sound right?

Reply:
This sounds familiar. So that's what I've done.

Reply:
Yes, your change is good.